File: app/src/main/java/com/commit451/nativestackblur/sample/MainActivity.java
package com.commit451.nativestackblur.sample;

import android.os.Bundle;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

import com.commit451.nativestackblur.sample.databinding.ActivityMainBinding;
import com.squareup.picasso.Picasso;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        String url = "https://images.unsplash.com/photo-1522648485144-849409408aee?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1180&q=80";
        Picasso.get()
                .load(url)
                .into(binding.image);
        Picasso.get()
                .load(url)
                .transform(new BlurTransformation(8))
                .into(binding.blurredImage);
    }
}

File: app/src/main/java/com/commit451/nativestackblur/sample/MainActivity.kt
package com.commit451.nativestackblur.sample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import coil.load
import com.commit451.nativestackblur.sample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    companion object {
        private const val ImageUrl =
            "https://images.unsplash.com/photo-1522648485144-849409408aee?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1180&q=80"
    }

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.image.load(ImageUrl)
        binding.blurredImage.load(ImageUrl) {
            transformations(BlurTransformation())
        }
    }
}